src/frontend/utils/csvDownload.ts

```diff
@@ -0,0 +1,81 @@
+import { state } from '../state'
+import { fieldName } from '../view/utils'
+
+export const getCurrentPageResults = () => {
+  const allPageResults =
+    state.searchResults?.slice(
+      state.skip,
+      state.skip + state.pagination.resultsPerPage
+    ) || []
+
+  const showFields = Object.entries(state.showFields)
+    .filter(([, show]) => show)
+    .map(([field]) => field)
+
+  const filteredFields = allPageResults.map((result) => {
+    const filteredResult = {}
+    for (const field of showFields) {
+      filteredResult[fieldName(field)] = result[field]
+    }
+    return filteredResult
+  })
+
+  return filteredFields
+}
+
+export const getAllResults = () => state.searchResults || []
+
+type DateValue = { value: string }
+
+export const convertToCSV = (data: { [key: string]: string | DateValue }[]) => {
+  if (data.length === 0) return ''
+
+  const headers = Object.keys(data[0]).join(',')
+  const rows = data.map((obj) => {
+    return Object.values(obj)
+      .map((value) => {
+        if (!value) {
+          return ''
+        }
+        if (typeof value === 'object') {
+          console.log(value)
+          value = (value as DateValue)?.value || value.toString()
+        }
+        return '"' + value.toString().replace(/"/g, '""') + '"'
+      })
+      .join(',')
+  })
+
+  return [headers, ...rows].join('\n')
+}
+
+export const generateFileDownloadFromCSVString = (
+  csv: string,
+  filename = 'data.csv'
```

Comment:
can you append "-current-results.csv" and "-all-results.csv" to the file name depending on what is chosen